src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -44,31 +46,160 @@ def gen_names(name):
         return name, name
 
 
-def update_dashboard_route(route_item, cluster_name, namespace):
-    metadata = route_item.get("generictemplate", {}).get("metadata")
-    metadata["name"] = f"ray-dashboard-{cluster_name}"
-    metadata["namespace"] = namespace
-    metadata["labels"]["odh-ray-cluster-service"] = f"{cluster_name}-head-svc"
-    spec = route_item.get("generictemplate", {}).get("spec")
-    spec["to"]["name"] = f"{cluster_name}-head-svc"
+# Check if the ingress api cluster resource exists
+def is_openshift_cluster():
+    try:
+        config_check()
+        api_instance = client.CustomObjectsApi(api_config_handler())
+        api_instance.get_cluster_custom_object(
+            "config.openshift.io", "v1", "ingresses", "cluster"
+        )
 
+        return True
+    except client.ApiException as e:
+        if e.status == 404:
+            return False
+        else:
+            print(f"Error detecting cluster type defaulting to Kubernetes: {e}")
+            return False
```

Comment:
As discussed earlier today. With KinD it seems to attempt to get ingresses in `config.openshift.io` API Group but it's forbidden. I think based on that attempt, it throws the error and defaults to vanilla (KinD) Kubernetes.

```
Error detecting cluster type defaulting to Kubernetes: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Audit-Id': '5c31d045-e2b0-4be1-8689-742a8f6fc976', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'X-Kubernetes-Pf-Flowschema-Uid': '4c0f4aec-2d7d-4e17-801f-02f507662410', 'X-Kubernetes-Pf-Prioritylevel-Uid': '9c711df4-3840-451f-bdf5-fd30956d147a', 'Date': 'Wed, 11 Oct 2023 18:21:17 GMT', 'Content-Length': '393'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"ingresses.config.openshift.io \"cluster\" is forbidden: User \"system:serviceaccount:test-ns-ghj2q:sdk-user\" cannot get resource \"ingresses\" in API group \"config.openshift.io\" at the cluster scope","reason":"Forbidden","details":{"name":"cluster","group":"config.openshift.io","kind":"ingresses"},"code":403}
```